src/main/kotlin/com/valbaca/gotlin/ch1/FetchAll.kt

```diff
@@ -1,26 +1,23 @@
 package com.valbaca.gotlin.ch1
 
 import kotlinx.coroutines.CoroutineScope
-import kotlinx.coroutines.Dispatchers
 import kotlinx.coroutines.channels.Channel
+import kotlinx.coroutines.coroutineScope
 import kotlinx.coroutines.launch
-import kotlinx.coroutines.runBlocking
 import org.http4k.client.ApacheClient
 import org.http4k.core.Method
 import org.http4k.core.Request
 import java.time.Duration.between
 import java.time.Instant.now
 
 /* p17 Fetch requests, concurrently https://github.com/adonovan/gopl.io/blob/master/ch1/fetchall/main.go */
-fun main(args: Array<String>) {
-    val urls = if (args.isNotEmpty()) {
-        args
-    } else {
+suspend fun main(args: Array<String>) {
```

Comment:
Kotlin supports the main method being a suspend method. The kotlin compiler will generate a main method that sets up the coroutine system and calls the suspending main.

Reply:
I just learned about this yesterday! :D